tmt/base.py

```diff
@@ -747,14 +828,14 @@ def __init__(
         # assign it to attribute *before* calling superclass and its handy
         # node key extraction.
         try:
-            directory = os.path.dirname(node.sources[-1])
-            relative_path = os.path.relpath(directory, node.root)
-            if relative_path == '.':
-                default_path = '/'
+            directory = Path(node.sources[-1]).parent
+            relative_path = directory.relative_to(Path(node.root))
+            if relative_path == Path.cwd():
```

Comment:
If the test metadata directory is the same as the `fmf` tree root then we get a `.`, right?

```
>>> pathlib.Path("/a/b/c").relative_to("/a/b/c")
PosixPath('.')
```

So we should not compare with `Path.cwd()`.

Reply:
I guess we need to compare with `Path(".")`, right?

```python
>>> Path('/a/b/c').relative_to(Path('/a/b/c')) == '.'
False

>>> Path('/a/b/c').relative_to(Path('/a/b/c')) == Path('.')
True
```

Reply:
I used `if relative_path == '.'` in 9eaf11a, shouldn't it be more or less what used to be in this file before my work?

Reply:
Before we were comparing `str` with `str` and now it's `Path` with `str`:

```python
>>> Path('.') == '.'
False
```

Or have I missed anything?

Reply:
Ah, now I get it. Fixed in c8c13e04.

Reply:
Good, thanks.